packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -85,6 +100,95 @@ class SemanticsTag {
   String toString() => '${objectRuntimeType(this, 'SemanticsTag')}($name)';
 }
 
+/// The result that contains the arrangement for the child
+/// [SemanticsConfiguration]s.
+///
+/// When the [PipelineOwner] builds the semantics tree, it uses the returned
+/// [ChildSemanticsConfigsResult] from
+/// [SemanticsConfiguration.childConfigsDelegate] to decide how semantics nodes
+/// should form.
+///
+/// Consider using [ChildSemanticsConfigsResultBuilder] to build the result.
+abstract class ChildSemanticsConfigsResult {
+  /// Gets [SemanticsConfiguration]s that are merging into parent semantics node.
+  ///
+  /// All merging up [SemanticsConfiguration]s that are either semantics
+  /// boundaries or are conflicting with other [SemanticsConfiguration]s
+  /// in the parent will form independent semantics nodes as usual. All other
+  /// [SemanticsConfiguration]s will be merged into parent semantics node.
```

Comment:
Returns the [SemanticsConfiguration]s that are supposed to be merged into the parent semantics node.

[SemanticsConfiguration]s that are either semantics boundaries or are conflicting with other [SemanticsConfiguration]s will form explicit semantics nodes. All others will be merged into the parent.